### DISCUSSION TITLE
how to get the older wz_mini_hacks versions

### DISCUSSION THREAD

**Discussion by vincom** [gtxaspec/wz_mini_hacks#695] [Q&A] [answered]
how to get the older wz_mini_hacks versions

**Comment by archandanime** [comment]
First, you need to clone the current Github repo:
```bash
> git clone https://github.com/gtxaspec/wz_mini_hacks
```

then change the working directory:
```bash
> cd wz_mini_hacks/
```

As we can see it is the current wz_mini_hacks:
```bash
> cat SD_ROOT/wz_mini/usr/bin/app.ver 
Sun Nov 19 11:58:58 PM PST 2023
```

To rollback to older wz_mini_hack version:

Browse the commit history: check [the main page](https://github.com/gtxaspec/wz_mini_hacks), click on the top right at "N Commits":

![Screenshot from 2023-12-30 18-01-35](https://github.com/gtxaspec/wz_mini_hacks/assets/36922812/d20a4f7c-48da-459d-9b9f-69d46f31af0c)

You would see commit history, wz_mini_hacks is changed after every new commit

For example, you want to rollback the cloned repo to the moment right after the below commit is pushed to the official repo, you have to obtain the commit's hash by clicking on the copy button on the right of the commit:

![Screenshot from 2023-12-30 18-05-36](https://github.com/gtxaspec/wz_mini_hacks/assets/36922812/5cbc2b0f-01aa-432f-b82d-261e20e4d14c)

Then you can rollback the cloned repo by running:
```bash
> git reset --hard <commit hash>
```

Now the cloned repo should be rolled back:
```bash
> cat SD_ROOT/wz_mini/usr/bin/app.ver
Fri Oct 13 04:58:00 PM PDT 2023
```

With the rolled back repo you can install the older wz_mini_hacks version.

**Comment by vincom** [comment, marked as answer]
thanks for the answer, but a little to complicated for me.......
but no worries as i thought i couldnt get the newest version working on any of my wyzev3 and panv2 working, problem was the damn sdcard i got for free, its a noname pos, a whole day wasted for that. win11 had no issues formatting or writing to it.
i always tell people dont rely on cheap sdcards if you value your data.
so it does go to show that the sdcard should be from a reputable brand
